### PR TITLE
Musicplayer-panel added. Started Musicplayer module in lua. Closes #32

### LINKED ISSUE #32: Implement music player
This is a very very subtle panorama element. It should just be a single button, which toggles music on and off, with a music note on it. When the music is on, it also displays the name/artist of the song playing next to it.

Have the currently playing song set by net tables, don't worry about the lua side for now.

### PR DESCRIPTION
**Pull request by Relacibo**
Created a Custom Panel, which can turn the Music on and off. Panel moves automatically, if minimapsize switches to extra large (Added same functionality to ngp-panel). Also displays Title and Artist, if the music is playing. Javascript listens on the Playertable musicplayer_music. The table has the
following entries: isMusicOn (integer), which is 1 if the music is on, otherwise 0.
musicTitle (String) and musicArtist (String). Fires the Event
('musicplayer_toggle') clientside. The goal would be to make the
musicplayer clientside. Idk if that can be implemented. Also started creating a lua-Module for the
musicplayer.
Closes #32

### CODE REVIEW

**Comment by Chronophylos** [comment]
why is `isMusicOn` not a `Boolean`?

**Comment by Relacibo** [comment]
Why is it not on the .gitignore-file?

**Comment by Chronophylos** [comment]
Because *.vpk is a compiled package that gets loaded by Dota. And *.vmap is the map you can actually edit.

**Comment by Relacibo** [comment]
I read the text on this file: https://github.com/bmddota/barebones/blob/source2/game/dota_addons/barebones/scripts/vscripts/libraries/playertables.lua .
and it seems that booleans are not supported.

**Comment by Chronophylos** [comment]
Boolean should work.

**Comment by Relacibo** [comment]
The problem is, I don't really know how the lua-side should be implemented.

File: content/panorama/scripts/custom_game/gold.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ var console = {
 var useFormatting = 'half';
 
 (function () {
-  PlayerTables.SubscribeNetTableListener('gold', onGoldChange);
+  //PlayerTables.SubscribeNetTableListener('gold', onGoldChange);
```
Comment (chrisinajar) [inline content/panorama/scripts/custom_game/gold.js]: I assume this isn't intentional..?

File: content/panorama/scripts/playertables/playertables_base.js [hunk 2]
```diff
@@ -229,7 +229,6 @@ function DeleteTableKeys(msg)
 
 (function(){
   GameUI.CustomUIConfig().PlayerTables = PlayerTables;
-
```
Comment (chrisinajar) [inline content/panorama/scripts/playertables/playertables_base.js]: similarly unnecessary change..?

File: game/scripts/vscripts/components/index.lua [hunk 3]
```diff
@@ -5,3 +5,4 @@ require('components/creeps/index')
 require('components/points/index')
 require('components/gold/index')
 require('components/zonecontrol/index')
+require('components/musicplayer/index')
```
Comment (chrisinajar) [inline game/scripts/vscripts/components/index.lua]: you should always have files end with a blank line

File: game/scripts/vscripts/components/musicplayer/index.lua [hunk 4]
```diff
@@ -0,0 +1,8 @@
+--[[
+
+musicplayercontrol
+
+Player can start and pause a song using a panorama-button.
+
+]]
+require('components/musicplayer/musicplayercontrol')
```
Comment (chrisinajar) [inline game/scripts/vscripts/components/musicplayer/index.lua]: no new line at end of file. we have a `.editorconfig`, if you're using sublime text there's a plugin to read it in. It'll do things like this automatically

File: game/scripts/vscripts/components/musicplayer/musicplayercontrol.lua [hunk 5]
```diff
@@ -0,0 +1,68 @@
+--[[
+  Author:
+    Relacibo
+]]
+
+if MusicPlayerControl == nil then
+  DebugPrint ( '[musicplayer/musicplayercontrol] Creating new MusicPlayerControl object' )
```
Comment (chrisinajar) [inline game/scripts/vscripts/components/musicplayer/musicplayercontrol.lua]: while you're in here you can remove the stuff the "`[musicplayer/musicplayercontrol]`" since the new debugging system automatically strips it out and replaces it with the filename and line number

**Comment by Relacibo** [comment]
@chrisinajar Thank you for the review. I'm working on it, it's not really finished yet.

File: game/scripts/vscripts/components/musicplayer/index.lua (hunk 4 above)
Comment (Relacibo) [inline game/scripts/vscripts/components/musicplayer/index.lua]: Ok, thank you. I switched to atom.io (from notepad++) and forgot to enable editorconf.

**Comment by warpdragon** [comment]
status? @Relacibo

**Comment by Relacibo** [comment]
Sorry, I'm not working on this anymore, but it was not possible for me to implement this. The dota-api doesn't seem to support it, but maybe there is a good workaround.

**Comment by warpdragon** [comment]
aight cool. thanks for the update.
